### PR TITLE
Add BookDto and CreateBookRequestDto with MapStruct mappers, implemen…

### PR DESCRIPTION
**Pull request by nklimovych**
…t BookController, update Service and Repository layers.

 - Added BookDto and CreateBookRequestDto as response and request DTOs respectively;
 - Implemented MapStruct to handle mapping between DTOs and entities;
 - Updated BookController with methods for retrieving all books, getting a book by ID, and creating a new book;
 - Implemented missing findById() method in BookService and BookRepository;
 - Created custom EntityNotFoundException class for handling non-existent entities;
 - Removed CommandLineRunner bean.

### CODE REVIEW

File: src/main/java/mate/academy/bookstore/repository/impl/BookRepositoryImpl.java [hunk 1]
```diff
@@ -46,7 +48,17 @@ public List<Book> findAll() {
         try (Session session = sessionFactory.openSession()) {
             return session.createQuery("FROM Book", Book.class).getResultList();
         } catch (Exception e) {
-            throw new DataProcessingException("Can't find all books", e);
+            throw new EntityNotFoundException("Can't find all books", e);
+        }
```
Comment (ghost) [inline src/main/java/mate/academy/bookstore/repository/impl/BookRepositoryImpl.java]: If an exception occurs while retrieving the books from DB, but there are some books there, will it be correct to tell the user that you can't find the books?

File: src/main/java/mate/academy/bookstore/repository/impl/BookRepositoryImpl.java [hunk 2]
```diff
@@ -46,7 +48,17 @@ public List<Book> findAll() {
...
+
+    @Override
+    public Book findById(Long id) {
+        try (Session session = sessionFactory.openSession()) {
+            Query<Book> query = session.createQuery("FROM Book WHERE id = :id", Book.class);
+            return query.setParameter("id", id).uniqueResult();
+        } catch (Exception e) {
+            throw new EntityNotFoundException("Can't find book by id: " + id, e);
         }
     }
```
Comment (ghost) [inline src/main/java/mate/academy/bookstore/repository/impl/BookRepositoryImpl.java]: Let's change this method to return an Optional<Book> and move the exception throwing to the service layer

**Review by ghost — CHANGES_REQUESTED** [review]
Good Job!
Check the comments

**Review by andrii-hoienko — APPROVED** [review]
Good job!

**Pull request merged**